File: desktop/src/lib/questionMedia.js
import { desktopApiBaseUrl } from './auth';

// Bazı (özellikle ehliyet) sorularında şıkların değeri düz metin değil, bir
// görsel bağlantısıdır. Değer tek parça bir URL/yol ve görsel uzantısıyla
// bitiyorsa görsel kabul edilir; aksi halde metindir.
export function isImageValue(value) {
  if (!value) return false;
  const v = String(value).trim();
  if (!v || /\s/.test(v)) return false;
  return /\.(png|jpe?g|webp|gif|svg|bmp)(\?.*)?(#.*)?$/i.test(v);
}

// Görsel yolunu tam URL'e çevirir. Mutlak http(s) linkleri (ör. jsdelivr CDN)
// olduğu gibi kullanılır; göreli yollar API tabanına eklenir.
export function buildQuestionImageUrl(path) {
  if (!path) return null;
  const v = String(path).trim();
  if (/^https?:\/\//i.test(v)) return v;
  return `${desktopApiBaseUrl}/${v.replace(/^\/+/, '')}`;
}

// Şık için ham "A) " önekini temizler.
export function stripOptionPrefix(option) {
  return String(option ?? '').replace(/^[A-F][).]\s*/i, '').trim();
}

File: mobile/lib/utils/question_media.dart
/// Soru şıkları için ortak görsel yardımcıları.
///
/// Bazı (özellikle ehliyet) sorularında şıkların değeri düz metin değil bir
/// görsel bağlantısıdır. Değer tek parça bir URL/yol ve görsel uzantısıyla
/// bitiyorsa görsel kabul edilir; aksi halde metindir.
bool isImageOptionValue(String? value) {
  final v = (value ?? '').trim();
  if (v.isEmpty || v.contains(RegExp(r'\s'))) return false;
  return RegExp(
    r'\.(png|jpe?g|webp|gif|svg|bmp)(\?.*)?(#.*)?$',
    caseSensitive: false,
  ).hasMatch(v);
}

/// Şıktaki baştaki "A) " / "A." önekini temizler.
String stripOptionPrefix(String? option) =>
    (option ?? '').replaceFirst(RegExp(r'^[A-Fa-f][).]\s*'), '').trim();
